### PR TITLE
Typing

### PR DESCRIPTION
**Pull request by Sytten**
### Description of changes

Adding a typing package to be published on NPM. We will then push a placeholder package in DefinitelTyped for `@types/llrt` which will re-export those types (this what Bun does to keep control over the typing in their repo).

### Checklist

- [x] Created unit tests in `tests/unit` and/or in Rust for my feature if needed
- [x] Ran `make fix` to format JS and apply Clippy auto fixes
- [x] Made sure my code didn't add any additional warnings: `make check`
- [x] Updated documentation if needed ([API.md](API.md)/[README.md](README.md)/Other)

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: types/buffer.d.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
+declare module "buffer" {
+  export type BufferEncoding = "utf8" | "base64" | "iso88591" | "hex";
```
Comment (richarddavison) [inline types/buffer.d.ts:2]: utf16le is also supported

**Review by richarddavison — CHANGES_REQUESTED** [review]
Outstanding work!!!

File: types/buffer.d.ts (hunk 1 above)
Comment (Sytten) [reply types/buffer.d.ts:2]: True that was added recently

**Review by richarddavison — APPROVED** [review]

**Review by richarddavison — APPROVED** [review]

**Comment by Sytten** [comment]
Waiting on #480 to be merged before I finish this

**Comment by Sytten** [comment]
Typing should be done now

**Review by richarddavison — APPROVED** [review]
Thanks for this amazing contribution!

**Pull request merged**